### DISCUSSION TITLE
Issue with matplotlib widgets in jupyter lab

### DISCUSSION THREAD

**Discussion by luikalu** [Mayitzin/ahrs#123] [Q&A]
Hi there, 
I really enjoy using this well documented library. 
However, somehow i cant get it to work with matplotlib widgets.
I get an error message when trying to plot something with matplotlib widgets enabled. 
Everything works fine without ahrs imported.

my code:

%matplotlib widget
from ahrs.filters import Mahony
import matplotlib.pyplot as plt
plt.plot([1,2,3])

This results in a type error and a deprecation warning.

I created a fresh environment using the latest version of ahrs.

Can you help me solve this problem?
Best regards,
Luis

[error_message.txt](https://github.com/user-attachments/files/17084277/error_message.txt)
[packages_in_environment.txt](https://github.com/user-attachments/files/17084278/packages_in_environment.txt)

**Comment by Mayitzin** [comment]
This is indeed a rare situation.

I have followed your error message and I cannot find any reason for `ahrs` to interfere with it.

It seems the origin of all that is the package `traitlets`.

I found a similar issue here: https://github.com/matplotlib/ipympl/issues/236

There, it seems when the user imports a package called `kornia` this funky warning happens, which he avoids by disabling the warnings. However, he later stopped getting those errors for some strange reason (https://github.com/kornia/kornia/issues/1425)

I certainly cannot find anything messing with the backend of `ipympl`. I would suggest the following to test the behavior:

- Change the line `%matplotlib widget` to `%matplotlib ipympl` to activate the `ipympl` backend.

If that doesn't work then:

- Import the required `ahrs` functions in the first lines, then continue with the code.

If none of that works, post again your error traces, and we check them again.

Regards,

Mario.

**Reply by luikalu** [reply]
Hey, thanks for the reply :)

I found two workarounds:
- version 3 of ahrs installed with pip works fine

-Importing ahrs after plotting something works:
%matplotlib widget
import matplotlib.pyplot as plt
plt.plot([1,2,3])
plt.show()
from ahrs.filters import Mahony
import ahrs
plt.plot([1,2,3,5,3])
plt.show()

I also tried different versions of traitlets, jupyter lab and ipykernel and changing the backend
Seems really weird. 

Best regards, 
Luis
